examples/react-full/renderer/pageContext.d.ts

```diff
@@ -0,0 +1,24 @@
+import 'vite-plugin-ssr';
+
+declare module "vite-plugin-ssr" {
+  namespace VitePluginSsr {
+    interface PageContextInit {
+      pageExports: {
+        documentProps?: {
+          title: string
+        }
+      }
```

Comment:
Where does `pageExports` comes from? That the init context and `renderPage` in `server.index.ts` complains that it is not there. I'm guessing that is something that is infered somewhere?

Reply:
Or it perhaps should be optional

Reply:
:thinking: I think there should be specific native handling for `pageExports` values. Currently they are defined in both `PageContextBuiltIn` and `OnBeforeRenderPageContextInternal`, and I think that is where they should be either merged or used differently.

EDIT: check my proposal in the PR where I defined a `PageExportsInternal` type

Reply:
> Where does `pageExports` comes from?

It's a `pageContext` built-int, see https://vite-plugin-ssr.com/pageContext

Reply:
So `pageContext.pageExports` is set by and comes from `vite-plugin-ssr`.

> check my proposal

Does it means I would need to do the following?

```ts
interface PageExports {
  documentProps?: {
    title: string
  }
}
```

Would it be possible to do what I tried instead?

```ts
interface PageContextInit {
  pageExports: {
    documentProps?: {
      title: string
    }
  }
}
```

Or do you just find it less user friendly?

I'm wondering, from the typings perspective it only makes a difference in the sense that `pageExports` is not required in `renderPage()`, am I right? Maybe a `Omit<PageContextInit, 'pageExports'>` for `renderPage()` would do the trick here.

Reply:
Currently with last commits you don't even need to define a `PageExports` or `PageContextInit['pageExports']`, as `PageContextBuiltIn['pageExports']` is now more specific.